mesonbuild/dependencies/blas_lapack.py

```diff
@@ -0,0 +1,774 @@
+            raise mesonlib.MesonException('Paths given for openblas_includedir and '
+                                          'openblas_librarydir in machine file must be absolute')
+        return has_dirs, [libdir, incdir]
+    elif incdir or libdir:
+        raise mesonlib.MesonException('Both openblas_includedir *and* openblas_librarydir '
+                                      'have to be set in your machine file (one is not enough)')
+    else:
+        raise mesonlib.MesonBugException('issue with openblas dependency detection, should not '
+                                         'be possible to reach this else clause')
+    return (False, [])
+
+
+class BLASLAPACKMixin():
+    def parse_modules(self, kwargs: T.Dict[str, T.Any]) -> None:
+        modules: T.List[str] = mesonlib.extract_as_list(kwargs, 'modules')
+        valid_modules = ['interface: lp64', 'interface: ilp64', 'cblas', 'lapack', 'lapacke']
+        for module in modules:
+            if module not in valid_modules:
+                raise mesonlib.MesonException(f'Unknown modules argument: {module}')
+
+        interface = [s for s in modules if s.startswith('interface')]
+        if interface:
+            if len(interface) > 1:
+                raise mesonlib.MesonException(f'Only one interface must be specified, found: {interface}')
+            self.interface = interface[0].split(' ')[1]
+        else:
+            self.interface = 'lp64'
+
+        self.needs_cblas = 'cblas' in modules
+        self.needs_lapack = 'lapack' in modules
+        self.needs_lapacke = 'lapacke' in modules
+
+    def check_symbols(self, compile_args, suffix=None, check_cblas=True,
+                      check_lapacke=True, lapack_only=False) -> None:
+        # verify that we've found the right LP64/ILP64 interface
+        symbols = []
+        if not lapack_only:
+            symbols += ['dgemm_']
+        if check_cblas and self.needs_cblas:
+            symbols += ['cblas_dgemm']
+        if self.needs_lapack:
+            symbols += ['zungqr_']
+        if check_lapacke and self.needs_lapacke:
+            symbols += ['LAPACKE_zungqr']
+
+        if suffix is None:
+            suffix = self.get_symbol_suffix()
+
+        prototypes = "".join(f"void {symbol}{suffix}();\n" for symbol in symbols)
+        calls = "  ".join(f"{symbol}{suffix}();\n" for symbol in symbols)
+        code = (f"{prototypes}"
+                "int main(int argc, const char *argv[])\n"
+                "{\n"
+                f"  {calls}"
+                "  return 0;\n"
+                "}"
+                )
+        code = '''#ifdef __cplusplus
+               extern "C" {
+               #endif
+               ''' + code + '''
+               #ifdef __cplusplus
+               }
+               #endif
+               '''
+
+        return self.clib_compiler.links(code, self.env, extra_args=compile_args)[0]
+
+    def get_variable(self, **kwargs: T.Dict[str, T.Any]) -> str:
```

Comment:
```suggestion
    def get_variable(self, **kwargs: T.Any) -> str:
```
splatting doesn't take a dict, it just takes a value type